internal/management/controller/database_controller.go

```diff
@@ -0,0 +1,393 @@
+	ctx context.Context,
+	database *apiv1.Database,
+	err error,
+) (ctrl.Result, error) {
+	oldDatabase := database.DeepCopy()
+	database.Status.Error = err.Error()
+	database.Status.Ready = false
+
+	var statusError *instance.StatusError
+	if errors.As(err, &statusError) {
+		// The body line of the instance manager contain the human
+		// readable error
+		database.Status.Error = statusError.Body
+	}
+
+	if err := r.Client.Status().Patch(ctx, database, client.MergeFrom(oldDatabase)); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{
+		RequeueAfter: databaseReconciliationInterval,
+	}, nil
+}
+
+// succeededReconciliation marks the reconciliation as succeeded
+func (r *DatabaseReconciler) succeededRenconciliation(
+	ctx context.Context,
+	database *apiv1.Database,
+) (ctrl.Result, error) {
+	oldDatabase := database.DeepCopy()
+	database.Status.Error = ""
+	database.Status.Ready = true
+	database.Status.ObservedGeneration = database.Generation
+
+	if err := r.Client.Status().Patch(ctx, database, client.MergeFrom(oldDatabase)); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{
+		RequeueAfter: databaseReconciliationInterval,
+	}, nil
+}
+
+// NewDatabaseReconciler creates a new databare reconciler
+func NewDatabaseReconciler(
+	mgr manager.Manager,
+	instance *postgres.Instance,
+) *DatabaseReconciler {
+	return &DatabaseReconciler{
+		Client:   mgr.GetClient(),
+		instance: instance,
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DatabaseReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&apiv1.Database{}).
+		Complete(r)
+}
+
+// GetCluster gets the managed cluster through the client
+func (r *DatabaseReconciler) GetCluster(ctx context.Context) (*apiv1.Cluster, error) {
+	var cluster apiv1.Cluster
+	err := r.Client.Get(ctx,
+		types.NamespacedName{
+			Namespace: r.instance.Namespace,
+			Name:      r.instance.ClusterName,
+		},
+		&cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return &cluster, nil
+}
+
+func (r *DatabaseReconciler) alignPgDatabase(ctx context.Context, obj *apiv1.Database) error {
+	db, err := r.instance.GetSuperUserDB()
+	if err != nil {
+		return fmt.Errorf("while connecting to the database: %w", err)
+	}
+
+	row := db.QueryRowContext(
+		ctx,
+		`
+		SELECT count(*)
+		FROM pg_database
+	        WHERE datname = $1
+		`,
+		obj.Spec.Name)
+	if row.Err() != nil {
+		return fmt.Errorf("while getting DB status: %w", err)
+	}
+
+	var count int
+	if err := row.Scan(&count); err != nil {
+		return fmt.Errorf("while getting DB status (scan): %w", err)
+	}
+
+	if count > 0 {
+		if err := r.patchDatabase(ctx, db, obj); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if err := r.createDatabase(ctx, db, obj); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *DatabaseReconciler) createDatabase(
+	ctx context.Context,
+	db *sql.DB,
+	obj *apiv1.Database,
+) error {
```

Comment:
This and the other functions that interact with Postgres, IMO should go in a dedicate file perhaps in `pkg/postgres`